civis/ml/_model.py

```diff
@@ -747,21 +778,30 @@ def from_existing(cls, train_job_id, train_run_id='latest', client=None):
         if p_id is None:
             warnings.warn('Model %s was trained with a newer version of '
                           'CivisML than is available in the API client '
-                          'version %s. Please update your API client version .'
+                          'version %s. Please update your API client version. '
                           'Attempting to use an older version of the '
                           'prediction code. Prediction will either fail '
                           'immediately or succeed.'
                           % (train_job_id, __version__), RuntimeWarning)
             p_id = max(_PRED_TEMPLATES.values())
         klass.predict_template_id = p_id
 
+        # The user might construct a model from an older version
+        # but still have access to the newest version, so we need
+        # to check the version separately in this function
+        is_newest_version = template_id == max(
+            _PRED_TEMPLATES.keys())
+        if is_newest_version:
```

Comment:
Is it possible to make this `if _NEWEST_CIVISML_VERSION`? If every switch looks the same, it'll be easier to search-and-replace when we need to update these blocks.

Reply:
The user could call `from_existing` before initializing any `ModelPipeline`s, so we can't assume that the global is defined. But I can refactor to use the `_set_template_version` method.

Do you think this logic is worth keeping after external release? If so, I might design some tests for this, if only to document the possible cases and expected behavior.

Reply:
You're calling the `ModelPipeline` constructor in the `cls` call above, so I think this method is covered.

The machinery you're adding here can apply to future release cycles, so I can see us keeping them.

Reply:
Okay, now I remember why I didn't use `_NEWEST_CIVISML_VERSION` here. The user might have access to the newest templates, but still be constructing a model from an older version of CivisML. This means that we need to test the version of the model, not the newest version the user has access to. It's testing something different than `_NEWEST_CIVISML_VERSION`.

Reply:
Is there a reason to set `_etl_train` here? It should already be set in the constructor. Does it matter if we set it even when users don't have access to v2 templates?

Reply:
We don't actually ever change `etl` or `_etl_train`, so I don't know that we need `_etl_train` at all.